include/envoy/server/bootstrap_extension_config.h

```diff
@@ -15,6 +15,12 @@ namespace Server {
 class BootstrapExtension {
 public:
   virtual ~BootstrapExtension() = default;
+
+  /**
+   * Called when server is done initializing and we have the ServerFactoryContext available.
+   * @param context general filter context through which persistent resources can be accessed.
+   */
+  virtual void serverInitialized(Configuration::ServerFactoryContext& context) PURE;
```

Comment:
name this `onServerInitialized`. The context can be still passed by constructor, because they are pointing to the same instance.

Reply:
The problem is that the Context is not fully initialized in the ctor, and I think it is bug prone (i.e. this bug is a good example) to do that. specifically, `clusterManager()` method in the context returns a _reference_ variable is nullptr; i.e. the user wouldn't even think to test it for nullptr as it is a reference.
To make it harder to write a bug, i think it is worth passing the bare minimum in the ctor (hence my change).

Reply:
Because that limits too much in createBootstrapExtension's access, there are more things should be made available for extension to be used, such as dispatcher/singletonManager/timeSource/Api etc. We can't just limit it to ValidationVisitor.